File: services/reservation/src/main/java/com/example/iter/reservation/dto/response/RentalHistoryResponse.java
package com.example.iter.reservation.dto.response;

import com.example.iter.auth.api.UserSummary;
import com.example.iter.reservation.api.RentalStatus;

import java.math.BigDecimal;
import java.time.LocalDate;

public record RentalHistoryResponse(
        Long rentalId,
        Long equipmentId,
        String equipmentName,
        String thumbnailUrl,
        UserSummary counterparty,
        LocalDate startDate,
        LocalDate endDate,
        BigDecimal totalPrice,
        RentalStatus status,
        int overdueDays
) {
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/dto/response/RentalHistoryResponse.kt
package com.example.iter.reservation.dto.response

import com.example.iter.auth.api.UserSummary
import com.example.iter.reservation.api.RentalStatus

import java.math.BigDecimal
import java.time.LocalDate

data class RentalHistoryResponse(
    val rentalId: Long?,
    val equipmentId: Long?,
    val equipmentName: String?,
    val thumbnailUrl: String?,
    val counterparty: UserSummary?,
    val startDate: LocalDate?,
    val endDate: LocalDate?,
    val totalPrice: BigDecimal?,
    val status: RentalStatus?,
    val overdueDays: Int,
) {
    // RentalHistoryMapper·RentalHistoryService(자바) 및 여러 테스트가 record 접근자
    // 스타일로 그대로 부른다.
    fun rentalId(): Long? = rentalId
    fun equipmentId(): Long? = equipmentId
    fun equipmentName(): String? = equipmentName
    fun thumbnailUrl(): String? = thumbnailUrl
    fun counterparty(): UserSummary? = counterparty
    fun startDate(): LocalDate? = startDate
    fun endDate(): LocalDate? = endDate
    fun totalPrice(): BigDecimal? = totalPrice
    fun status(): RentalStatus? = status
    fun overdueDays(): Int = overdueDays
}
